src/lcm/simulation/simulate.py

```diff
@@ -519,11 +530,11 @@ def _calculate_discrete_argmax(values, choice_axes):
 # ======================================================================================
 
 
-def dict_product(d):
+def dict_product(d: dict[str, Array]) -> tuple[dict[str, Array], int]:
```

Comment:
Maybe include in the func name that the len is returned, too? Also, a short explainer of what exactly is returned would be great. I honestly have no idea from the code. Instead of `entries`, use `values`.